src/index.js

```diff
@@ -21,42 +21,113 @@ import {
 
 export default class ImgixClient {
   constructor(opts = {}) {
-    this.settings = { ...DEFAULT_OPTIONS, ...opts };
+    this._settings = { ...DEFAULT_OPTIONS, ...opts };
     // a cache to store memoized srcset width-pairs
     this.targetWidthsCache = {};
-    if (typeof this.settings.domain != 'string') {
+    if (typeof this._settings.domain != 'string') {
       throw new Error('ImgixClient must be passed a valid string domain');
     }
 
-    if (DOMAIN_REGEX.exec(this.settings.domain) == null) {
+    if (DOMAIN_REGEX.exec(this._settings.domain) == null) {
       throw new Error(
         'Domain must be passed in as fully-qualified ' +
           'domain name and should not include a protocol or any path ' +
           'element, i.e. "example.imgix.net".',
       );
     }
 
-    if (this.settings.includeLibraryParam) {
-      this.settings.libraryParam = 'js-' + ImgixClient.version();
+    if (this._settings.includeLibraryParam) {
+      const defaultLibraryParam = 'js-' + ImgixClient.version();
+      this._settings.libraryParam =
+        this._settings.libraryParam || defaultLibraryParam;
     }
 
-    this.settings.urlPrefix = this.settings.useHTTPS ? 'https://' : 'http://';
+    this._settings.urlPrefix = this._settings.useHTTPS ? 'https://' : 'http://';
   }
 
   static version() {
     return VERSION;
   }
 
+  get settings() {
+    return this._settings;
+  }
+  get domain() {
+    return this._settings.domain;
+  }
+
+  set domain(value) {
+    if (typeof value != 'string') {
+      throw new Error('ImgixClient must be passed a valid string domain');
+    }
+
+    if (DOMAIN_REGEX.exec(value) == null) {
+      throw new Error(
+        'Domain must be passed in as fully-qualified ' +
+          'domain name and should not include a protocol or any path ' +
+          'element, i.e. "example.imgix.net".',
+      );
+    }
+    this._settings.domain = value;
+  }
+
+  get includeLibraryParam() {
+    return this._settings.includeLibraryParam;
+  }
+
+  set includeLibraryParam(value) {
+    if (typeof value !== 'boolean') {
+      throw 'includeLibraryParam must be a boolean';
+    }
+    this._settings.includeLibraryParam = value;
+  }
+
+  get libraryParam() {
+    return this._settings.libraryParam;
+  }
+
+  set libraryParam(value) {
+    if (typeof value !== 'string') {
+      throw 'libraryParam must be string';
+    }
+    this._settings.libraryParam = value;
+  }
```

Comment:
Open question: should we signal to users that they can explicitly set this?
I know we wanted to enable it for use in the plugins but I'm worried about openly signaling this in the API. Maybe we can replace it with an underscored `set libraryParam`?